components/SearchResult.tsx

```diff
@@ -1,39 +1,37 @@
-import { useQuery } from "@tanstack/react-query";
+import { useInfiniteQuery } from "@tanstack/react-query";
 import Result from "./Result";
 import { Album, SearchResponseSchema } from "@/schemas/collections.schemas";
 import ResultLoading from "./ResultLoading";
 
 export default function SearchResult({ searchValue }: { searchValue: string }) {
-  const { data, isError, isLoading } = useQuery({
-    queryKey: ["search", searchValue],
-    queryFn: async () => {
-      const params = new URLSearchParams({
-        search: searchValue,
-      });
-      const res = await fetch("/api/search?" + params);
-      const data = await res.json();
-      const parsedData = SearchResponseSchema.parse(data);
+  const { data, error, isFetching, fetchNextPage, hasNextPage } =
+    useInfiniteQuery({
+      queryKey: ["search", searchValue],
+      queryFn: async ({ pageParam }) => {
+        const params = new URLSearchParams({
+          search: searchValue,
+          page: String(pageParam),
+        });
+        const res = await fetch("/api/search?" + params);
+        const data = await res.json();
+        const parsedData = SearchResponseSchema.parse(data);
+        return parsedData;
+      },
+      initialPageParam: 1,
+      getNextPageParam: (lastPage) => {
+        if (lastPage.pagination.page < lastPage.pagination.pages) {
```

Comment:
this should always just return `lastPage.pagination.page + 1`

the check for whether it's valid to call a next page or not should happen earlier in the code (show/hiding the "more results" button)